src/Symfony/Component/Security/Core/Authentication/Provider/DaoAuthenticationProvider.php

```diff
@@ -56,7 +57,7 @@ protected function checkAuthentication(UserInterface $user, UsernamePasswordToke
         $currentUser = $token->getUser();
         if ($currentUser instanceof UserInterface) {
             if ($currentUser->getPassword() !== $user->getPassword()) {
-                throw new BadCredentialsException('The credentials were changed from another session.');
+                throw new AccountPasswordChangedException('The credentials were changed from another session.');
```

Comment:
The `DaoAuthenticationProvider` can be extended or used elsewhere, thus can break user-land implementations :cry: